src/pint/models/binary_piecewise.py

```diff
@@ -0,0 +1,325 @@
+    All the detailed calculations are in the stand alone BTmodel.
+    The aim for this class is to connect the stand alone binary model with PINT platform
+    BTmodel special parameters:
+    GAMMA Binary Einsten delay coeeficient
+    """
+
+    register = True
+
+    def __init__(self):
+        super(BinaryBTPiecewise, self).__init__()
+        self.binary_model_name = "BT_piecewise"
+        self.binary_model_class = BTpiecewise
+        self.add_param(
+            floatParameter(
+                name="GAMMA",
+                value=0.0,
+                units="second",
+                description="Time dilation & gravitational redshift",
+            )
+        )
+        self.A1_value_funcs = []
+        self.T0_value_funcs = []
+        self.remove_param("M2")
+        self.remove_param("SINI")
+        self.T0.value = 1
+        self.A1.value = 1
+        self.add_group_range(50000, 51000, frozen=False, j=0)
+        self.add_piecewise_param("T0", "d", None, 0)
+        self.add_piecewise_param("A1", "ls", None, 0)
+        # self.set_special_params(["T0X_0000", "PieceLowerBound_0000", "PieceUpperBound_0000"])
+
+    def add_group_range(self, group_start_mjd, group_end_mjd, frozen=True, j=None):
+        # print('hello from add range')
+        if group_end_mjd is not None and group_start_mjd is not None:
+            if group_end_mjd < group_start_mjd:
+                raise ValueError("Starting MJD is greater than ending MJD.")
+            elif j < 0:
+                raise ValueError(
+                    f"Invalid index for group: {j} should be greater than or equal to 0"
+                )
+            elif j > 9999:
+                raise ValueError(
+                    f"Invalid index for group. Cannot index beyond 9999 (yet?)"
+                )
+
+        i = f"{int(j):04d}"
+        self.add_param(
+            prefixParameter(
+                name="PLB_{0}".format(i),
+                units="MJD",
+                unit_template=lambda x: "MJD",
+                description="Beginning of paramX interval",
+                description_template=lambda x: "Beginning of paramX interval",
+                parameter_type="MJD",
+                time_scale="utc",
+                value=group_start_mjd,
+            )
+        )
+        self.add_param(
+            prefixParameter(
+                name="PUB_{0}".format(i),
+                units="MJD",
+                unit_template=lambda x: "MJD",
+                description="End of paramX interval",
+                description_template=lambda x: "End of paramX interval",
+                parameter_type="MJD",
+                time_scale="utc",
+                value=group_end_mjd,
+            )
+        )
+        # print("going to binary_piecewise setup")
+        # self.setup()
+        # self.validate()
+
+    def update_binary_object(self, toas=None, acc_delay=None):
+        super().update_binary_object(toas=toas, acc_delay=acc_delay)
```

Comment:
If all this method does is call the same method on `super()` can it not be omitted completely? Several other methods here seem to be in the same situation.